interview-questions.md

```diff
@@ -7,63 +7,80 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a group of statements that preform a certain task.
 
-  Researched answer:
+  Researched answer: Functions are reusable pieces of code that are usually called upon a lot. They also
+  can have information pushed through what we call an argument.
 
 
 
 2. What is the difference between .map() and .filter()?
 
-  Your answer:
+  Your answer: A .map goes through an array and returns one with the same length. A .filter
+  filters out some type of value.
 
-  Researched answer:
+  Researched answer: So .map goes through an array and returns another one with the
+  the same length, .filter does almost the same thing but only returns the
+  ones with true statement that you decided.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log(), is like your "go" button because without it a lot of arrays won't be able to run.
+  A return is used when you input something into you console but will only return what you tell it to.
 
-  Researched answer:
+  Researched answer:console.log(), is almost you could say a go button because it
+  itll print things to the console so it can start the code. A return wont start
+  your console and only return something you put into it.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument in a function is a way to pass information in the function.
 
-  Researched answer:
+  Researched answer: An argument contains outside information that a function needs in order to execute properly.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Its when one person is writing the code while the other can try to help with useful information
+  and so  that two head can take a look at a code and get two different mind trying to figure out the same
+  problem.
 
-  Researched answer:
+  Researched answer: One is the driver with a switch off when the the pair seem fit. Usually after a few challenges
+  or if one partner wants to show there solution or try at the challenge. While one is driving the other can give ideas or opinions on what todo on when writing the code.
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer: Tdd is when you write out a test for your code then type it in and see why it was to fail.
```

Comment:
Kind of, but lets again take these interview questions as if an interviewer just asked you to describe the proper flow of TDD and what it actually is. You need to make sure you say something like: "TDD stands for Test Driven Development. It follows the methodology of Red-Green-Refactor where you write the test first, see it fail, then write the code and see it pass, then refactor it as needed."